### PR TITLE
Test for `/api/security/users` endpoint

### PR DESCRIPTION
**Pull request by olenagerasimova**
Part of #965 
Updated management-api, asto, http and added IT for `/api/security/users` endpoint.
Also, due to `NoSuchMethod` runtime exception with hamcrest `Matchers.allOff` method in one of the tests, added hamcrest dependency.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -113,8 +113,8 @@ SOFTWARE.
     </site>
   </distributionManagement>
   <properties>
-    <maven.compiler.source>17</maven.compiler.source>
-    <maven.compiler.target>17</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
```
Comment (genryxy) [inline pom.xml:117]: why version was downgraded?

File: src/test/java/com/artipie/api/ArtipieApiITCase.java [hunk 2]
```diff
@@ -69,6 +69,20 @@ void dashboardIsUp() throws IOException {
...
 
+    @Test
+    void getUsers() throws IOException {
+        this.deployment.assertExec(
+            "Artipie failed to return list of the users",
+            new ContainerResultMatcher(
+                Matchers.is(0),
+                new StringContainsInOrder(
+                    new ListOf<String>("\"name\":\"alice\"", "\"name\":\"bob\"")
+                )
```
Comment (genryxy) [inline src/test/java/com/artipie/api/ArtipieApiITCase.java:80]: is the order for names in this line determined?

**Review by genryxy — CHANGES_REQUESTED** [review]
@olenagerasimova thanks, two questions

**Comment by olenagerasimova** [comment]
@genryxy thanks

**Review by genryxy — DISMISSED** [review]
@olenagerasimova thanks, looks good to me
@g4s8 please check/merge

File: pom.xml [hunk 3]
```diff
@@ -404,6 +409,16 @@ SOFTWARE.
...
+      <plugins>
+          <plugin>
+              <groupId>org.apache.maven.plugins</groupId>
+              <artifactId>maven-compiler-plugin</artifactId>
+              <configuration>
+                  <source>11</source>
+                  <target>11</target>
+              </configuration>
+          </plugin>
+      </plugins>
```
Comment (g4s8) [inline pom.xml:421]: Why Java lang version changed?

**Review by g4s8 — CHANGES_REQUESTED** [review]
@olenagerasimova just one comment

File: pom.xml (hunk 3 above)
Comment (olenagerasimova) [reply pom.xml]: my bad, sorry

**Comment by olenagerasimova** [comment]
@g4s8 thanks, fixed

**Comment by olenagerasimova** [comment]
@g4s8 please, merge

**Comment by olenagerasimova** [comment]
@g4s8 ping

**Review by genryxy — APPROVED** [review]

**Review by g4s8 — APPROVED** [review]
@olenagerasimova thanks

**Pull request merged**